@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -858,17 +906,15 @@ export class TextElementsRenderer {
 
     private sortTextElements(textElements: TextElement[], maxViewDistance: number) {
         const distancePriorityFactor = 0.1;
-        const indexPriorityFactor = 0.01 * (1 / textElements.length);
+        const uidPriorityFactor = 0.0000000001;
 
         // Compute the sortPriority once for all elements, because the computation is done more
         // than once per element. Also, make sorting stable by taking the index into the array into
         // account, this is required to get repeatable results for testing.
-        for (let i = 0; i < textElements.length; i++) {
-            const textElement = textElements[i];
-
+        for (const textElement of textElements) {
             textElement.sortPriority =
                 textElement.priority +
-                i * indexPriorityFactor +
+                textElement.priority * uidPriorityFactor +
```

Comment:
I believe this is a mistake
```suggestion
                textElement.uid * uidPriorityFactor +
```